File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        int h = 0;
        int w = 0;
        
        for(int[] size : sizes) {
            h = Math.max(h, Math.max(size[0],size[1]));
            w = Math.max(w, Math.min(size[0],size[1]));
        }
        
        return h*w;
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int max_w = 0;
    int max_h = 0;
        
    for(auto &size : sizes){
        max_w = max(max_w, max(size[0], size[1]));
        max_h = max(max_h, min(size[0], size[1]));
    }
    
    return max_w * max_h;
    
}